Extension/src/LanguageServer/configurations.ts

```diff
@@ -987,7 +987,7 @@ export class CppProperties {
                 }
             } else {
                 // However, if compileCommands are used and compilerPath is explicitly set, it's still necessary to resolve variables in it.
-                if (configuration.compilerPath === "${default}") {
+                if (configuration.compilerPath === "${default}" && settings.defaultCompilerPath) {
```

Comment:
This ` && settings.defaultCompilerPath` should be removed (which becomes valid after adding `| null` to the type of `Configuration.compilerPath`.